### PR TITLE
Container and Image migration for aufs

### PR DESCRIPTION
**Pull request by crosbymichael**

### CODE REVIEW

File: graphdriver/aufs/aufs.go [hunk 1]
```diff
@@ -95,18 +95,25 @@ func supportsAufs() error {
...
+func (AufsDriver) String() string {
 	return "aufs"
 }
 
-func (d *AufsDriver) Status() [][2]string {
+func (AufsDriver) Status() [][2]string {
 	return nil
 }
 
+func (a AufsDriver) Exists(id string) bool {
```
Comment (shykes) [inline graphdriver/aufs/aufs.go]: Comment please :)

File: graphdriver/aufs/migrate.go [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+}
+
+func pathExists(pth string) bool {
+	if _, err := os.Stat(pth); err != nil {
+		return false
+	}
+	return true
+}
+
+// Migrate existing images and containers from docker < 0.7.x
```
Comment (shykes) [inline graphdriver/aufs/migrate.go:26]: Could you give a high-level summary of how the migration works? While it's still fresh in our minds.

**Comment by vieux** [comment]
LGTM

File: graphdriver/aufs/migrate.go (hunk 2 above)
Comment (crosbymichael) [inline graphdriver/aufs/migrate.go:26]: @shykes done

**Comment by shykes** [comment]
Question, if the move fails and we fallback to a symlink - next time will the migration re-try the move to get rid of the symlink. That would be great - that way over time we eventually converge towards a full migration.

**Comment by crosbymichael** [comment]
@shykes 

Updated

**Pull request merged**